python/python/lance/lance/__init__.pyi

```diff
@@ -76,3 +76,6 @@ class LanceFileMetadata:
     num_global_buffer_bytes: int
     global_buffers: List[LanceBufferDescriptor]
     columns: List[LanceColumnMetadata]
+
+class _Session:
+    def size_bytes(self) -> int: ...
```

Comment:
For clarity, should it be `cache_size_bytes()`?

Reply:
I think, if we have any other stateful objects in the session, we would want to capture that here as well.  In other words, it would make more sense to rename `Session` to `DatasetCache` than it would to rename `size_bytes` to `cache_size_bytes`.

Reply:
I see. For context, I'm thinking other methods we might set on the context will be execution args, like:

```python
class Session:
    def set_cpu_pool_size(num_cpus: int): ...
    def set_io_pool_size(num_cpus: int): ...
    def set_memory_limit(memory_limit: int): ...
    def set_spill_location(path: Path): ...
```

So I'm thinking of the Session object as containing more state than the cache.

Reply:
> So I'm thinking of the Session object as containing more state than the cache.

Sorry, my comment was probably confusing.  I think this is good, my point should have been "it doesn't make sense to rename Session to DatasetCache and therefore doesn't make sense to rename size_bytes to cache_size_bytes".

For example, maybe DF starts to need some "thread local temp space" or "slab allocator working area" or whatever (I'm probably just making terms up at this point).  This would go in the session, would not be trivial in size, and would not be considered a "cache", but we would still want to make sure that `size_bytes` includes it.